### ISSUE TITLE
Dataloss when Spyder gets confused about which file goes with which editor tab

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1269]
_From [kinnison...@gmail.com](https://code.google.com/u/104685941974947235171/) on 2013-02-12T10:00:14Z_

Spyder Version:  2.2.0beta1
Python Version:  2.7.3
Qt Version:      4.8.1, PyQt4 (API v1) 4.9.1 on Linux 

What steps will reproduce the problem?

 1. see issue `#1225` for the basic setup
2. I've not yet tried to reproduce, I've not seen a clear pattern yet. What is the expected output? What do you see instead? After using an undocked editor for a while, throughout normal development, I sometimes see the undocked editor get out of sync with the docked editor.  Some of the file tabs in the docked editor (which I only use rarely) will show the wrong names for the file actually being displayed. 

Please provide any additional information below

. I used to think that this was just a minor issue, not even worth mentioning separate from issue `#1225` .  But earlier I almost lost work due to this bug.  The undocked editor was out of sync, so I decided to restart Spyder.  I was asked to save one of the files I was working on, and chose to save it (obvious choice).  When I reopened Spyder, I found that my file had been overwritten with the contents of a different file.

It seems like this isn't just a visual mismatch, but that Spyder is really loosing track of which file is which because it doesn't properly separate the two editors.  I only managed to recover my work because it was in Dropbox.  Otherwise, I'd have lost everything back to my last commit.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1269_

**Comment by spyder-issues-migrator** [comment]
_From [kinnison...@gmail.com](https://code.google.com/u/104685941974947235171/) on 2013-02-12T08:34:45Z_

Still not able to reproduce regularly, but it seems like it could be related to the handling of untitled#.py documents.  Those always seem to be involved when the problem arises (from what I can remember).

**Comment by spyder-issues-migrator** [comment]
_From [kinnison...@gmail.com](https://code.google.com/u/104685941974947235171/) on 2013-02-14T07:05:15Z_

It happened again. I had a number of files open, closed one of them, and then noticed that one of the remaining tabs in the docked editor only was incorrectly displaying the contents of the file I just closed.  Also, this bug is not carried over in saved sessions (I tried "save session and quit" to see if I could save the buggy state, but everything was working properly when I reloaded it).
